comp/dogstatsd/listeners/uds_linux.go

```diff
@@ -121,12 +123,12 @@ func getEntityForPID(pid int32, capture bool) (string, error) {
 
 // entityForPID returns the entity ID for a given PID. It can return
 // errNoContainerMatch if no match is found for the PID.
-func entityForPID(pid int32, capture bool) (string, error) {
+func entityForPID(pid int32, capture bool, wmeta optional.Option[workloadmeta.Component]) (string, error) {
 	if capture {
 		return replay.ContainerIDForPID(pid)
 	}
 
-	cID, err := metrics.GetProvider().GetMetaCollector().GetContainerIDForPID(int(pid), pidToEntityCacheDuration)
+	cID, err := metrics.GetProvider(wmeta).GetMetaCollector().GetContainerIDForPID(int(pid), pidToEntityCacheDuration)
```

Comment:
I don't see any method "GetProvider" which take an optional.Option[workloadmeta.Component] as an argument but I may have not seen it

Reply:
thxx